packages/components/src/InputFile/InputFile.tsx

```diff
@@ -190,30 +197,49 @@ export function InputFile({
   }
 
   async function uploadFile(file: File) {
-    const { url, key = uuidv1(), fields = {} } = await getUploadParams(file);
+    const {
+      url,
+      key = uuidv1(),
+      fields = {},
+      httpMethod = "POST",
+    } = await getUploadParams(file);
 
     const fileUpload = getFileUpload(file, key);
     onUploadStart && onUploadStart({ ...fileUpload });
 
-    const formData = new FormData();
-    Object.entries(fields).forEach(([field, value]) =>
-      formData.append(field, value),
-    );
-    formData.append("file", file);
-
-    axios
-      .post(url, formData, {
-        headers: { "X-Requested-With": "XMLHttpRequest" },
-        onUploadProgress: progressEvent =>
-          onUploadProgress &&
-          onUploadProgress({
-            ...fileUpload,
-            progress: progressEvent.loaded / progressEvent.total,
-          }),
-      })
-      .then(() => {
-        onUploadComplete && onUploadComplete({ ...fileUpload, progress: 1 });
-      });
+    const uploadProgressUpdate = (progressEvent: any) => {
```

Comment:
nit: we try to name those callbacks passed in `onCallback` props with `handleCallback` that way, it's easier to connect the dots just by looking at it.

```suggestion
    const handleUploadProgress = (progressEvent: any) => {
```